App.js

```diff
@@ -13,59 +13,84 @@ import styles from "./styles";
 
 const rnSentianceEmitter = new NativeEventEmitter(RNSentiance);
 
+const userLinkFlag = 'SDK_USER_LINKED';
+
 export default class App extends Component {
   state = {
     userId: "...",
     sdkVersion: "...",
     userActivityText: "...",
+    userLinkInstallId: "...",
     data: [],
-    subscriptionsAdded: false
   };
 
   async componentDidMount() {
     try {
-      await this.setupSdk();
-      this.subscribeAfterSDKSetup();
+      this.subscribeSDKEvents();
+      await this.initSDK();
       await this.requestPermissionAndroid();
     } catch (e) {
       console.error(e.code, e);
     }
   }
 
   componentWillUnmount() {
-    if (this.state.subscriptionsAdded) {
-      this.unSubscribe();
-    }
+    this.unSubscribe();
   }
 
   unSubscribe() {
     this.sdkStatusSubscription.remove();
     this.sdkUserActivityUpdateSubscription.remove();
+    this.userLinkListener.remove();
+  }
+
+  /**
+   * The user-linking logics are required to be implemented in your backend.
+   * 1. Sending the installId to your backend.
+   * 2. Calling the Sentiance API with the installId and the externalId in your backend.
+   * 3. Responding the result.
+   * If the backend responds successful, call RNSentiance.userLinkCallback(true);. Otherwise, RNSentiance.userLinkCallback(false);
+   * SDK will not be initialized if linking is failed.
+   */
+  async linkMetaUser(installId) {
+    RNSentiance.userLinkCallback(true);
```

Comment:
@countcain If I remember correctly, calling `userLinkCallback` directly here crashed the app on iOS. Can we find out why that happens, and if it's possible to avoid it? Because as far as the flow goes, invoking the callback with `true` should proceed normally, but fail initialization due to no linking.

Additionally, I would propose explicitly throwing an exception here with a message saying that user linking has not been implemented.

Reply:
Yes, directly calling `RNSentiance.userLinkCallback(true);` will crash iOS SDK and result in `LINK_FAILED` error thrown from Android SDK.

Without `userLinkCallback` calling or calling it with `false` will fail the initialization. We can throw an error here to remind them to implement the user linking.